harvester_e2e_tests/integration/test_networks.py

```diff
@@ -0,0 +1,919 @@
+        delete_vm(api_client, unique_name, wait_timeout)
+
+    @pytest.mark.p0
+    @pytest.mark.dependency(name="vlan_mgmt_connection")
+    def test_vlan_to_mgmt_connection(self, api_client, request, client, unique_name,
+                                     image_info, vlan_network, wait_timeout):
+        """
+        Manual test plan reference:
+        https://harvester.github.io/tests/manual/network/edit-network-form-change-management-to-vlan/
+
+
+        Steps:
+        1. Create an external VLAN network
+        2. Create a new VM
+        3. Make sure that the network is set to the vlan network with bridge as the type
+        4. Wait until the VM boot in running state
+        5. Edit VM and change from external VLAN to management network
+        6. Check VM should save and reboot
+        7. Check can ping VM with management network from Harvester node
+        8. Check can SSH to VM with management network from Harvester node
+        9. Check can't SSH to VM with management network from external host
+        """
+
+        vip = request.config.getoption('--endpoint').strip('https://')
+
+        node_password = request.config.getoption("--host-password")
+
+        image_url = image_info.url
+
+        # Check image exists
+        code, data = api_client.images.get(image_name)
+
+        if code == 404:
+            create_image_url(api_client, image_name, image_url, wait_timeout)
+
+        self.ssh_client(client, vip, node_user, node_password,
+                        tcp, wait_timeout)
+
+        self.ssh_client(client, vip, node_user, node_password,
+                        restart_ssh, wait_timeout)
+
+        spec = api_client.vms.Spec(1, 2, mgmt_network=False)
+        spec.user_data += cloud_user_data.format(password=vm_credential["password"])
+        unique_name = unique_name + "-vlan-mgmt"
+
+        # Create VM
+        spec.add_image(image_name, "default/" + image_name)
+        spec.add_network("default", vlan_network['id'])
+
+        code, data = api_client.vms.create(unique_name, spec)
+        assert 201 == code, (f"Failed to create vm with error: {code}, {data}")
+
+        # Check VM start in running state
+        check_vm_running(api_client, unique_name, wait_timeout)
+
+        # Check until VM ip address exists
+        check_vm_ip_exists(api_client, unique_name, wait_timeout)
+
+        # get data from running VM and transfer to spec
+        code, data = api_client.vms.get(unique_name)
+        assert 200 == code, (f"Failed to get specific vm content: {code}, {data}")
+        spec = spec.from_dict(data)
+
+        # Switch to vlan network
+        spec.mgmt_network = True
+
+        # Update VM spec
+        # code, data = api_client.vms.update(unique_name, spec)
+        # assert 200 == code, (f"Failed to update specific vm with spec: {code}, {data}")
+
+        # Remove external vlan network from spec
+        net_uid = vlan_network['id']
+        spec.networks = [net for net in spec.networks
```

Comment:
In here, we can simply do:
 ```python
spec.networks = []
spec.mgmt_network = True
```
rather than
```python
spec.mgmt_network = True
spec.networsk = [...]
```

https://github.com/harvester/tests/pull/681#discussion_r1188698247